### PR TITLE
Fix redirect after delete resource without index

### LINKED ISSUE #626: question: Can I hide dashboards from the left menu but keep them in as relationships?
I really only want to see `projects` and `users` on the left. But each model has a bunch of relationships. I find if I remove the dashboard file, I can remove the option in the menu, but then I cannot display that relationship when viewing a project or user.

### PR DESCRIPTION
**Pull request by edimossilva**
fix #626

**Steps to reproduce**
- `Resourse A` has many `Resource B` 
- `Resourse B` has no index page
- go to show page of `Resourse A` and click on destroy of `Resource B` 
- resource is destroyed

**Behavior before this fix**
- **An error is raised**
image

**Behavior after this fix**
- **is redirected `Resourse A` show page**

### CODE REVIEW

**Comment by edimossilva** [comment]
@pablobm

**Review by pablobm — COMMENTED** [review]
Thank you @edimossilva!

I see why you had to create a new model `MetaTagGroups`. The way the example app is put together is not great. I'm currently working on a branch that will bring some order to this.

A few days before you, someone created a related PR: https://github.com/thoughtbot/administrate/pull/1973. I think yours may complement it. I feel that, instead of creating this new query param `origin_url`, perhaps the same effect could be achieved by overriding `after_resource_destroyed_path`, as provided by the other PR. What do you think? If you think it makes sense, we should wait until the other PR is merged before rebasing yours on top.

**Comment by edimossilva** [comment]
> Thank you @edimossilva!
> 
> I see why you had to create a new model `MetaTagGroups`. The way the example app is put together is not great. I'm currently working on a branch that will bring some order to this.
> 
> A few days before you, someone created a related PR: #1973. I think yours may complement it. I feel that, instead of creating this new query param `origin_url`, perhaps the same effect could be achieved by overriding `after_resource_destroyed_path`, as provided by the other PR. What do you think? If you think it makes sense, we should wait until the other PR is merged before rebasing yours on top.

Hi @pablobm , PR #1973 looks nice. I agree that the best way is overriding `after_resource_destroyed_path`. Let's wait for the merge of #1973 and rebase this one.

About the new model `MetaTagGroups`. Should I wait for your branch as well?

**Comment by pablobm** [comment]
I'd say let's wait for #1973 first and then let's see where my branch is when it arrives, and if we can use it to simplify your spec.

**Comment by pablobm** [comment]
This is the branch that reorganises the example app: https://github.com/thoughtbot/administrate/pull/1998. When it's merged, we can revisit this PR.

File: app/controllers/administrate/application_controller.rb
Comment (pablobm) [inline app/controllers/administrate/application_controller.rb:84]: This code assumes that `after_resource_destroyed_path` always returns a hash with an `:action` key. This will probably not be the case when someone overrides it. (And it was created specifically to be overridden).

We need some other way to tell if the route exists. Perhaps we can do something with `url_for`? (No idea from the top of my head).

Alternative idea: only provide `origin_url` if the show page doesn't exist. This means checking the existence of the route when rendering the link, instead of when following the link. Then we always follow `origin_url` if present.

**Review by pablobm — CHANGES_REQUESTED** [review]
Hello @edimossilva; apologies for the late review; I have very little time these days. I'm still trying to catch up with the Administrate backlog, but it will take some time.

I think there's an issue with your PR (sorry...). Can you see if you can improve on that? I'll try to be more responsive on this PR in the future.

**Comment by pablobm** [comment]
Closing due to lack of activity.

**Pull request closed without merging**
